dask/dataframe/tests/test_multi.py

```diff
@@ -796,6 +796,23 @@ def test_concat_with_operation_remains_hlg():
     assert_eq(result, expected)
 
 
+def test_concat_dataframe_empty():
+    df = pd.DataFrame({"a": [100, 200, 300]}, dtype="int64")
+    empty_df = pd.DataFrame([], dtype="int64")
+    df_concat = pd.concat([df, empty_df])
+
+    ddf = dd.from_pandas(df, npartitions=1)
+    empty_ddf = dd.from_pandas(empty_df, npartitions=1)
+    ddf_concat = dd.concat([ddf, empty_ddf])
+    assert_eq(df_concat, ddf_concat)
+
+    empty_df_with_col = pd.DataFrame([], columns=["x"], dtype="int64")
+    df_concat_with_col = pd.concat([df, empty_df_with_col])
+    empty_ddf_with_col = dd.from_pandas(empty_df_with_col, npartitions=1)
+    ddf_concat_with_col = dd.concat([ddf, empty_ddf_with_col])
+    assert_eq(df_concat_with_col, ddf_concat_with_col)
```

Comment:
Something more like this? (Test passes)
 
```suggestion
    # NOTE: empty_ddf_with_col.index.dtype == dtype('O') while ddf.index.dtype == dtype('int64')
    # when we concatenate the resulting ddf_concat_with_col.index.dtype == dtype('O'). However,
    # the pandas version is smart enough to identify the empty rows and assign dtype('int64'), 
    # which causes assert_eq to fail when checking dtype. Hence the follwoing line. 
    ddf_concat_with_col._meta.index = ddf_concat_with_col._meta.index.astype('int64')

    assert_eq(df_concat_with_col, ddf_concat_with_col, check_dtype=False)
```